relay-protocol/src/traits.rs

```diff
@@ -125,6 +125,103 @@ pub trait IntoValue: Debug + Empty {
     }
 }
 
+/// Hides an iterator of [`Getter`]s that is used to iterate over arbitrary [`Getter`]
+/// implementations.
+///
+/// # Example
+///
+/// ```
+/// use relay_protocol::{Getter, GetterIter, Val};
+///
+/// struct Exception {
```

Comment:
_nit_: Following the example of `Getter`, could we make these slightly more generic like `Root` and `Nested`? That way, the protocol documentation is not too tied to the specific event schema.